datafusion/src/optimizer/constant_folding.rs

```diff
@@ -247,6 +248,25 @@ impl<'a> ExprRewriter for ConstantRewriter<'a> {
                     }
                 }
             }
+            Expr::Cast {
+                expr: inner,
+                data_type,
+            } => match inner.as_ref() {
+                Expr::Literal(val) => {
+                    let scalar_array = val.to_array();
```

Comment:
Great idea👍 I think converting to an array is the best we can do for now, without duplicating the code.